File: duna-annotations/src/main/java/io/duna/port/Port.java
package io.duna.port;

import javax.inject.Qualifier;
import java.lang.annotation.*;

/**
 * Annotates extensions defining ports to the services cluster.
 *
 * @author <a href="mailto:cemelo@redime.com.br">Carlos Eduardo Melo</a>
 */
@Qualifier
@Documented
@Target({ElementType.TYPE, ElementType.FIELD, ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
public @interface Port {
}

File: duna-core/src/main/kotlin/io/duna/core/external/Port.kt
package io.duna.core.external

import com.google.inject.BindingAnnotation

/**
 * Annotates extensions defining ports to the services cluster.
 *
 * @author [Carlos Eduardo Melo][cemelo@redime.com.br]
 */
@BindingAnnotation
@MustBeDocumented
@Target(AnnotationTarget.CLASS, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class Port
